opencog/atoms/base/Handle.h

```diff
@@ -263,6 +268,17 @@ struct content_based_atom_ptr_less
     }
 };
 
+struct content_based_handle_less
+{
+    bool operator()(const Handle& hl, const Handle& hr) const
+    {
+        return Handle::content_based_atoms_less(hl.const_atom_ptr(),
+                                                hr.const_atom_ptr());
+    }
```

Comment:
Why are you passing Atom* here? why not just pass in the handles themselves?  Its kind-of dangerous to pass around raw, naked Atom* -- which is why the API mostly doesn't have Atom*'s in it anywhere -- except for this new usage.